boto3/session.py

```diff
@@ -78,7 +81,10 @@ def __init__(
 
         if aws_access_key_id or aws_secret_access_key or aws_session_token:
             self._session.set_credentials(
-                aws_access_key_id, aws_secret_access_key, aws_session_token
+                aws_access_key_id,
+                aws_secret_access_key,
+                aws_session_token,
+                aws_account_id,
```

Comment:
What happens if only `aws_account_id` is set? Do we fold this into the other resolved credentials?

Reply:
The input is currently ignored and the client will fall back to one of the credential providers. All of the existing credential values can be individually set on a boto3 session and a client can be created successfully even if the other credential values are empty.
```python
from boto3.session import Session
session = Session(aws_access_key_id='foo')
s3 = session.client('s3')
print(s3._request_signer._credentials.__dict__)
{'access_key': 'foo', 'secret_key': None, 'token': None, 'method': 'explicit'}
```
It seemed wrong to extend this behavior to account ID. This is similar to how `botocore.session.Session.create_client` [handles explicitly set credential values](https://github.com/boto/botocore/blob/develop/botocore/session.py#L942). If only the session token is set here, it will be ignored and use a credential provider instead.

It's not the best experience to silently ignore an input so maybe we could add validation here or in `set_credentials` instead.